---
layout: page
title: Publications
group: navigation
permalink: "publications.html"
---
{% include JB/setup %}

<script type='text/javascript' src='https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js'></script>

Publication Repositories
------------------------

[Google Scholar Citations](http://scholar.google.com/citations?user=8XOPyM4AAAAJ&hl=en) |
[PubMed](http://www.ncbi.nlm.nih.gov/pubmed/?term=Hansen%2C+Kasper+D%5BAuthor%5D+or+Hansen%2C+Kasper+Daniel%5BAuthor%5D+or+17681996%5BPmid%5D+or+21177974%5BPmid%5D+or+23999529%5BPmid%5D+or+25285208%5BPmid%5D) | 
[ORCID](http://orcid.org/0000-0003-0086-0687): 0000-0003-0086-0687 |
[Europe PMC Citations](http://europepmc.org/authors/0000-0003-0086-0687)

Unpublished Preprints
---------------------

A Nellore, AE Jaffe, <b>JP Fortin</b>, J Alquicira-Hernández, L Collado-Torres, S Wang, RA Phillips, N Karbhari, <b>KD Hansen</B>, B Langmead<sup>&dagger;</sup>, and JT Leek<sup>&dagger;</sup>.<br>
<i>Human splicing diversity across the Sequence Read Archive.</i><br>
bioRxiv 2016.<br>
[DOI](http://dx.doi.org/10.1101/038224) | <span data-badge-type="4" data-doi="10.1101/038224"" data-hide-no-mentions="true" class="altmetric-embed"> </span><br />

<div data-badge-type="4" data-doi="10.1101/038224"" data-hide-no-mentions="true" class="altmetric-embed"></div><br />

A Nellore<sup>&dagger;</sup>, C Wilks, <b>KD Hansen</b>, JT Leek, and B Langmead<sup>&dagger;</sup>.<br>
<i>Rail-dbGaP: a protocol and tool for analyzing protected genomic data in a commercial cloud.</i><br>
bioRxiv 2015.<br>
[DOI](http://dx.doi.org/10.1101/035287) | <span data-badge-type="4" data-doi="10.1101/035287"" data-hide-no-mentions="true" class="altmetric-embed"></span><br>


Computational Biology
---------------------

L Kannan, M Ramos, A Re, N El-Hachem, A Safikhani, DMA Gendoo, S Davis, S Gomez-Cabrero, R Castelo, <b>KD Hansen</b>, VJ Carey, M Morgan, AC Culhane, B Haibe-Kains, and L Waldron<sup>&dagger;</sup>.<br>
<i>Public data and open source tools for multi-assay genomic investigation of disease.</i><br>
Brief Bioinformatics 2015, In press.<br>
[DOI](http://dx.doi.org/10.1093/bib/bbv080) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/26463000) |
<span data-badge-type="4" data-doi="10.1093/bib/bbv080"" data-hide-no-mentions="true" class="altmetric-embed"></span><br>

L Gatto, <b>KD Hansen</b>, MR Hoopmann, H Henning, O Kohlbacher, and A Beyer<sup>&dagger;</sup>.<br>
<i>Testing and validation of computational methods for mass spectrometry.</i><br>
Journal of Proteome Research 2016, 15(3):809-814.<br>
[DOI](http://dx.doi.org/10.1021/acs.jproteome.5b00852) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/26549429) |
<span data-badge-type="4" data-doi="10.1021/acs.jproteome.5b00852"" data-hide-no-mentions="true" class="altmetric-embed"></span><br>

AR Vandiver, A Idrizi, L Rizzardi, AP Feinberg, and <b>KD Hansen</b><sup>&dagger;</sup>.<br>
<i>DNA methylation is stable during replication and cell cycle arrest.</i><br>
Scientific Reports 2015, 5:17911.<br>
[DOI](http://dx.doi.org/10.1038/srep17911) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/26648411) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4673417) | 
<span data-badge-type="4" data-doi="10.1038/srep17911"" data-hide-no-mentions="true" class="altmetric-embed"></span><br>

A Pacis, L Tailleux, AM Morin, J Lambourne, JL Maclsaac, V Yotova, A Dumaine, A Danckaert, F Luca, J Grenier, <b>KD Hansen</b>, B Gicquel, M Yu, A Pai, C He, J Tung, T Pastinen, MS Kobor, R Pique-Regi, Y Gilad<sup>&dagger;</sup>, and LB Barreiro<sup>&dagger;</sup>.<br>
<i>Bacterial infection remodels the DNA methylation landscape of human dendritic cells.</i><br>
Genome Research 2015, 25(12):1801-1811.<br>
[DOI](http://dx.doi.org/10.1101/gr.192005.115) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/26392366) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4665002) |
<span data-badge-type="4" data-doi="10.1101/gr.192005.115"" data-hide-no-mentions="true" class="altmetric-embed"></span><br>

<b>JP Fortin</b> and <b>KD Hansen</b><sup>&dagger;</sup>.<br>
<i>Reconstructing A/B compartments as revealed by Hi-C using long-range correlations in epigenetic data.</i><br>
Genome Biology 2015, 16:180.<br>
[DOI](http://www.dx.doi.org/10.1186/s13059-015-0741-y) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/26316348) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4574526) |
<span data-badge-type="4" data-doi="10.1186/s13059-015-0741-y"" data-hide-no-mentions="true" class="altmetric-embed"></span><br>
<font size=-1>
Research highlight by Mourad and Cuvier in [Genome Biology](http://www.dx.doi.org/10.1186/s13059-015-0752-8)
</font>

AR Vandiver, RA Irizarry, <b>KD Hansen</b>, LA Garza, A Runarsson, Arni, X Li, AL Chien, TS Wang, SG Leung, S Kang, and AP Feinberg<sup>&dagger;</sup>.<br>
<i>Age and sun exposure-related widespread genomic blocks of hypomethylation in nonmalignant skin.</i><br>
Genome Biology 2015, 16:80.<br>
[DOI](http://www.dx.doi.org/10.1186/s13059-015-0644-y) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/25886480) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4423110) |
<span data-badge-type="4" data-doi="10.1186/s13059-015-0644-y"" data-hide-no-mentions="true" class="altmetric-embed"></span>

X Hong<sup>&lowast;</sup>, K Hao<sup>&lowast;</sup>, C Ladd-Acosta<sup>&lowast;</sup>, <b>KD Hansen</b>, Tsai X, HJ Liu, X Xu, TA Thornton, D Caruso, CA Keet, Y Sun, G Wang, W Luo, R Kumar, R Fuleihan, AM Singh, Kim RE, JS Story, RS Gupta, P Gao, Z Chen, SO Walker, TR Bartell, TH Beaty, MD Fallin, R Schleimer, PG Holt, Nadeau KC, RA Wood, JA Pongracic, DE  Weeks, and X Wang<sup>&dagger;</sup>.<br>
<i>Genome-wide association study identifies peanut allergy-specific loci and evidence of epigenetic mediation in US children.</i><br>
Nature Communications 2015, 6:6304.<br>
[DOI](http://dx.doi.org/10.1038/ncomms7304) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/25710614) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4340086) |
<span data-badge-type="4" data-doi="10.1038/ncomms7304"" data-hide-no-mentions="true" class="altmetric-embed"></span>

MJ Ziller, <b>KD Hansen</b>, A Meissner<sup>&dagger;</sup>, MJ Aryee<sup>&dagger;</sup>.<br>
<i>Coverage recommendations for methylation analysis by whole-genome bisulfite sequencing.</i><br>
Nature Methods 2015, 12(3):230-232.<br>
[DOI](http://dx.doi.org/10.1038/nmeth.3152) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/25362363) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4344394) |
<span data-badge-type="4" data-doi="10.1038/nmeth.3152"" data-hide-no-mentions="true" class="altmetric-embed"></span>

W Huber<sup>&dagger;</sup>, VJ Carey, R Gentleman, S Anders, M Carlson, BS Carvalho, H
Corrada-Bravo, S Davis, L Gatto, T Girke, R Gottardo, Florian Hahne, <b>KD Hansen</b>, RA Irizarry,
M Lawrence, MI Love, J MacDonald, V Obenchain, AK Oleś, H Pagès, A Reyes, P Shannon, GK Smyth, D
Tenenbaum, L Waldron, and M Morgan.<br>
<i>Orchestrating high-throughput genomic analysis with Bioconductor.</i><br>
Nature Methods 2015, 12(2):115-121.<br>
[DOI](http://dx.doi.org/10.1038/nmeth.3252) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/25633503) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4509590) |
<span data-badge-type="4" data-doi="10.1038/nmeth.3252"" data-hide-no-mentions="true" class="altmetric-embed"></span>

<b>JP Fortin</b>, A Labbe, M Lemire, BW Zanke, TJ Hudson, EJ Fertig, CMT Greenwood and <b>KD Hansen</b><sup>&dagger;</sup>.<br>
<i>Functional normalization of 450k methylation array data improves replication in large cancer studies.</i><br>
Genome Biology 2014, 15:503.<br>
[DOI](http://dx.doi.org/10.1186/s13059-014-0503-2) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/25599564) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4283580) |
<span data-badge-type="4" data-doi="10.1186/s13059-014-0503-2"" data-hide-no-mentions="true" class="altmetric-embed"></span>

HT Bjornsson<sup>&lowast;,&dagger;</sup>, JS Benjamin<sup>&lowast;</sup>, L Zhang, J Weissman, EE Gerber, YC Chen, RG Vaurio, MC Potter, <b>KD Hansen</b>, HC Dietz.<br>
<i>Histone deacetylase inhibition rescues structural and functional brain deficits in a mouse model of Kabuki syndrome.</i><br>
Science Translational Medicine 2014, 6(256):256ra135.<br>
[DOI](http://dx.doi.org/10.1126/scitranslmed.3009278) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/25273096) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4406328) |
<span data-badge-type="4" data-doi="10.1126/scitranslmed.3009278" data-hide-no-mentions="true" class="altmetric-embed"></span>

<b>JP Fortin</b><sup>&dagger;</sup>, EJ Fertig, <b>KD Hansen</b><sup>&dagger;</sup>.<br>
<i>shinyMethyl: interactive quality control of Illumina 450k DNA methylation arrays in R.</i><br>
F1000Research 2014, 3:175.<br>
[DOI](http://dx.doi.org/10.12688/f1000research.4680.2) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/25285208) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4176427) |
<span data-badge-type="4" data-doi="10.12688/f1000research.4680.2" data-hide-no-mentions="true" class="altmetric-embed"></span>

MJ Aryee, AE Jaffe, H Corrada-Bravo, C Ladd-Acosta, AP Feinberg, <b>KD Hansen</b><sup>&dagger;</sup>, RA Irizarry<sup>&dagger;</sup>.<br>
<i>Minfi: A flexible and comprehensive Bioconductor package for the analysis of Infinium DNA Methylation microarrays.</i><br>
Bioinformatics 2014, 30(10):1363-1369.<br>
[DOI](http://dx.doi.org/10.1093/bioinformatics/btu049) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/24478339) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4016708) |
<span data-badge-type="4" data-doi="10.1093/bioinformatics/btu049" data-hide-no-mentions="true" class="altmetric-embed"></span>

AC Frazee, S Sabunciyan, <b>KD Hansen</b>, RA Irizarry, JT Leek<sup>&dagger;</sup>.<br>
<i>Differential expression analysis of RNA-seq data at single-base resolution.</i><br>
Biostatistics 2014, 15(3):413-426.<br>
[DOI](http://dx.doi.org/10.1093/biostatistics/kxt053) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/24398039) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4059460) |
<span data-badge-type="4" data-doi="10.1093/biostatistics/kxt053" data-hide-no-mentions="true" class="altmetric-embed"></span>

C Ladd-Acosta, <b>KD Hansen</b>, E Briem, MD Fallin, WE Kaufmann, AP Feinberg<sup>&dagger;</sup>.<br>
<i>Common DNA methylation alterations in multiple brain regions in autism.</i><br>
Molecular Psychiatry 2014, 19(8):862-871.<br>
[DOI](http://dx.doi.org/10.1038/mp.2013.114) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/23999529) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4184909) |
<span data-badge-type="4" data-doi="10.1038/mp.2013.114" data-hide-no-mentions="true" class="altmetric-embed"></span><br>
<font size=-1>
Faculty of 1000 reviews [f1000](http://f1000.com/prime/718097865)<br>
</font>

<b>KD Hansen</b><sup>&lowast;</sup>, S Sabunciyan<sup>&lowast;</sup>, B Langmead, N Nagy, R Curley, G Klein, E Klein, D Salamon, AP Feinberg<sup>&dagger;</sup>.<br>
<i>Large-scale hypomethylated blocks associated with Epstein-Barr virus-induced B-cell
immortalization.</i><br>
Genome Research 2014, 24(2):177-184.<br>
[DOI](http://dx.doi.org/10.1101/gr.157743.113) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/24068705) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3912409) |
<span data-badge-type="4" data-doi="10.1101/gr.157743.113" data-hide-no-mentions="true" class="altmetric-embed"></span>

ML Smith<sup>&dagger;</sup>, KA Baggerly, H Bengtsson, ME Ritchie, <b>KD Hansen</b><sup>&dagger;</sup>.<br>
<i>illuminaio: An open source IDAT parsing tool for Illumina microarrays.</i><br>
F1000Research 2013, 2:264.<br>
[DOI](http://dx.doi.org/10.12688/f1000research.2-264.v1) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/24701342) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3968891) |
<span data-badge-type="4" data-doi="10.12688/f1000research.2-264.v1" data-hide-no-mentions="true" class="altmetric-embed"></span>

<b>KD Hansen</b><sup>&lowast;,&dagger;</sup>, B Langmead<sup>&lowast;&dagger;</sup>, RA Irizarry<sup>&dagger;</sup>.<br>
<i>BSmooth: from whole genome bisulfite sequencing reads to
differentially methylated regions.</i><br>
Genome Biology 2012, 13:R83.<br>
[DOI](http://dx.doi.org/10.1186/gb-2012-13-10-r83) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/23034175) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3491411) |
<span data-badge-type="4" data-doi="10.1186/gb-2012-13-10-r83" data-hide-no-mentions="true" class="altmetric-embed"></span>
<br>
<font size=-1>
Highly accessed
</font>

BR Herb<sup>&lowast;</sup>, F Wolschin<sup>&lowast;</sup>, <b>KD Hansen</b>, MJ Aryee, B Langmead,
RA Irizarry, GV Amdam<sup>&dagger;</sup>, AP Feinberg<sup>&dagger;</sup>.<br>
<i>Reversible switching between epigenetic states in honeybee behavioral subcastes.</i><br>
Nature Neuroscience 2012, 15(10):1371-1373.<br>
[DOI](http://dx.doi.org/10.1038/nn.3218) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/22983211) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3518384) |
<span data-badge-type="4" data-doi="10.1038/nn.3218" data-hide-no-mentions="true" class="altmetric-embed"></span>
<br>
<font size=-1>
Coverage by Nature News [URL](http://www.nature.com/news/job-swapping-makes-its-mark-on-honeybee-dna-1.11418)<br>
Coverage by LA Times [URL](http://www.latimes.com/news/science/sciencenow/lat-sci-sn-worker-bees-switch-jobs-via-their-epigenomes-20120917,0,3764312.story)<br>
Coverage by The New Republic [URL](http://www.tnr.com/blog/plank/107327/epigenetics-bees-biology)<br>
Coverage by the Guardian UK [URL](http://www.guardian.co.uk/environment/2012/sep/16/bee-study-behaviour)<br>
Coverage by Discover Magazine [URL](http://blogs.discovermagazine.com/notrocketscience/2012/09/16/bee-epigenetics-nurses-foragers-reversible/)<br>
Coverage by The Scientist [URL](http://the-scientist.com/2012/09/16/controlling-bee-fate/)<br>
</font>

J Tung<sup>&dagger;</sup>, LB Barreiro, ZP Johnson, <b>KD Hansen</b>, V Michopoulos, D
Toufexis, K Michelini, ME Wilson, Y Gilad<sup>&dagger;</sup>.<br>
<i>Social environment is associated with gene regulatory variation in
the rhesus macaque immune system.</i><br>
PNAS 2012, 109(17):6490-6495.<br>
[DOI](http://dx.doi.org/10.1073/pnas.1202734109) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/22493251) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3340061) |
<span data-badge-type="4" data-doi="10.1073/pnas.1202734109" data-hide-no-mentions="true" class="altmetric-embed"></span>
<br>
<font size=-1>
Faculty of 1000 reviews [f1000](http://f1000.com/14267420)<br>
Coverage by NY times [URL](http://www.nytimes.com/2012/04/10/science/changes-in-social-status-seen-in-monkeys-genes.html?src=rechp)<br>
Coverage by Genomeweb [URL](http://www.genomeweb.com/blog/healthy-popularity)<br>
Coverage by Videnskab.db [URL](http://videnskab.dk/krop-sundhed/taber-aber-har-darligt-immunforsvar)
</font>

S Munshaw, HS Hwang, M Torbenson, J Quinn, <b>KD Hansen</b>, J
Astemborski, SH Mehta, SC Ray, DL Thomas, A Balagopal<sup>&dagger;</sup>.<br>
<i>Laser captured hepatocytes show association of association of butyrylcholinesterase gene loss and
fibrosis progression in hepatitis C infected drug users.</i><br>
Hepatology 2012, 56(2):544-554.<br>
[DOI](http://dx.doi.org/10.1002/hep.25655) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/22331678) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3388175) |
<span data-badge-type="4" data-doi="10.1002/hep.25655" data-hide-no-mentions="true" class="altmetric-embed"></span>

<b>KD Hansen</b>, RA Irizarry, and Z WU<sup>&dagger;</sup>.<br>
<i>Removing technical variability in RNA-Seq data using conditional
quantile normalization.</i><br>
Biostatistics 2012, 13(2):204-216.<br>
[DOI](http://dx.doi.org/10.1093/biostatistics/kxr054) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/22285995) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3297825) |
<span data-badge-type="4" data-doi="10.1093/biostatistics/kxr054" data-hide-no-mentions="true" class="altmetric-embed"></span>

<b>KD Hansen</b><sup>&lowast;</sup>, W Timp<sup>&lowast;</sup>, H
Corrada Bravo<sup>&lowast;</sup>, S Sabunciyan<sup>&lowast;</sup>, B
Langmead<sup>&lowast;</sup>, OG McDonald, B Wen, H Wu, Y Liu, D Diep, E Briem, K Zhang, RA
Irizarry<sup>&dagger;</sup>, and AP Feinberg<sup>&dagger;</sup>.<br>
<i>Increased methylation variation in epigenetic domains across cancer
types.</i><br>
Nature Genetics 2011, 43:768-775.<br>
[DOI](http://dx.doi.org/10.1038/ng.865) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/21706001) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3145050) |
<span data-badge-type="4" data-doi="10.1038/ng.865" data-hide-no-mentions="true" class="altmetric-embed"></span>
<br>
<font size=-1>
Faculty of 1000 reviews [f1000](http://f1000.com/11598956)<br>
Comment by JP Issa in Nat Genet, <i>Epigenetic variation and cellular Darwinism</i>.
[DOI](http://dx.doi.org/10.1038/ng.897) 
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/21792236)<br>
Described in Cell
[DOI](http://dx.doi.org/10.1016/j.cell.2011.09.021)<br>
Coverage by Genomeweb [URL](http://www.genomeweb.com/sequencing/study-suggests-widespread-loss-epigenetic-regulation-cancer-genomes)
</font>

<b>KD Hansen</b>, Z Wu, RA Irizarry<sup>&dagger;</sup>, and JT Leek<sup>&dagger;</sup>.<br>
<i>Sequencing technology does not eliminate biological
variability.</i><br>
Nature Biotechnology 2011, 29:572-573.<br>
[DOI](http://dx.doi.org/10.1038/nbt.1910) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/21747377) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3137276) |
<span data-badge-type="4" data-doi="10.1038/nbt.1910" data-hide-no-mentions="true" class="altmetric-embed"></span>

AN Brooks<sup>&lowast;</sup>, L Yang<sup>&lowast;</sup>, MO Duff, <b>KD Hansen</b>, JW Park, S Dudoit, SE
Brenner<sup>&dagger;</sup>, and BR Graveley<sup>&dagger;</sup>.<br>
<i>Conservation of an RNA regulatory map between Drosophila and
mammals.</i><br>
Genome Research 2011, 21(2):193-202.<br>
[DOI](http://dx.doi.org/10.1101/gr.108662.110) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/20921232) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3032923) |
<span data-badge-type="4" data-doi="10.1101/gr.108662.110" data-hide-no-mentions="true" class="altmetric-embed"></span>

The modENCODE Consortium, <i> et al.</i><br>
<i>Identification of functional elements and regulatory circuits by
Drosophila modENCODE.</i><br>
Science 2010, 330:1787-1797.<br>
[DOI](http://dx.doi.org/10.1126/science.1198374) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/21177974) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3192495) |
<span data-badge-type="4" data-doi="10.1126/science.1198374" data-hide-no-mentions="true" class="altmetric-embed"></span>

B Langmead, <b>KD Hansen</b>, and JT Leek<sup>&dagger;</sup>.<br>
<i>Cloud-scale RNA-sequencing differential expression analysis with
Myrna.</i><br>
Genome Biology 2010, 11:R83.<br>
[DOI](http://dx.doi.org/10.1186/gb-2010-11-8-r83) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/20701754) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2945785) |
<span data-badge-type="4" data-doi="10.1186/gb-2010-11-8-r83" data-hide-no-mentions="true" class="altmetric-embed"></span>
<br>
<font size=-1>
Highly accessed
</font>

<b>KD Hansen</b><sup>&dagger;</sup>, SE Brenner, and S Dudoit.<br>
<i>Biases in Illumina transcriptome sequencing caused by random
hexamer priming.</i><br>
Nucleic Acids Research 2010, 38(12):e131.<br>
[DOI](http://dx.doi.org/10.1093/nar/gkq224) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/20395217) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2896536) |
<span data-badge-type="4" data-doi="10.1093/nar/gkq224" data-hide-no-mentions="true" class="altmetric-embed"></span>

JH Bullard<sup>&lowast;,&dagger;</sup>, EA Purdom<sup>&lowast;</sup>, <b>KD Hansen</b>, and
S Dudoit.<br>
<i>Evaluation of statistical methods for normalization and
differential expression in mRNA-Seq experiments.</i><br> 
BMC Bioinformatics 2010, 11:94.<br>
[DOI](http://dx.doi.org/10.1186/1471-2105-11-94) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/20167110) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2838869) |
<span data-badge-type="4" data-doi="10.1186/1471-2105-11-94" data-hide-no-mentions="true" class="altmetric-embed"></span>
<br>
<font size=-1>
Highly accessed
</font>

<b>KD Hansen</b><sup>&lowast;</sup>, LF Lareau<sup>&lowast;</sup>, M Blanchette, RE Green, Q Meng,
J Rehwinkel, FL Gallusser, E Izaurralde, DC Rio, S Dudoit, and SE Brenner<sup>&dagger;</sup>. <br>
<i>Genome-wide identification of alternative splice forms
down-regulated by nonsense-mediated mRNA decay in Drosophila.</i><br>
PLoS Genetics 2009, 5(6):e1000525.<br>
[DOI](http://dx.doi.org/10.1371/journal.pgen.1000525) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/19543372) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2689934) |
<span data-badge-type="4" data-doi="10.1371/journal.pgen.1000525" data-hide-no-mentions="true" class="altmetric-embed"></span>

A Lee<sup>&lowast;</sup>, <b>KD Hansen</b><sup>&lowast;</sup>, J Bullard<sup>&lowast;</sup>, S
Dudoit, and G Sherlock<sup>&dagger;</sup>.<br>
<i>Novel low abundance and transient RNAs in yeast revealed by tiling microarrays and ultra
high-throughput sequencing are not conserved across closely related yeast species</i>.<br> 
PLoS Genetics 2008, 4(12):e1000299.<br>
[DOI](http://dx.doi.org/10.1371/journal.pgen.1000299) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/19096707) |
[PMC](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2601015) |
<span data-badge-type="4" data-doi="10.1371/journal.pgen.1000299" data-hide-no-mentions="true" class="altmetric-embed"></span>


Editorials (not peer-reviewed)
------------------------------

W Huber<sup>&dagger;</sup>, V Carey, S Davis, <b>KD Hansen</b> and M Morgan.<br>
<i>The Bioconductor channel in F1000Research.</i><br>
F1000Research 2015, 4.<br>
[DOI](http://dx.doi.org/10.12688/f1000research.6758.1)


Published Preprints
-------------------

A Pacis, L Tailleux, J Lambourne, V Yotova, A Dumaine, A Danckaert, F Luca, JC Grenier, <b>KD Hansen</b>, B Gicquel, M Yu, A Pai, J Tung, C He, T Pastinen, R Pique-Regi, Y Gilad<sup>&dagger;</sup> and L Barreiro<sup>&dagger;</sup>.<br>
<i>Bacterial Infection Remodels the DNA Methylation Landscape of Human Dendritic Cells.</i><br>
bioRxiv 2015.<br>
[DOI](http://dx.doi.org/10.1101/016022) |
Published in Genome Research 2015.

<b>JP Fortin</b> and <b>KD Hansen<sup>&dagger;</sup>.</b><br>
<i>Reconstructing A/B compartments as revealed by Hi-C using long-range correlations in epigenetic data.</i><br>
bioRxiv, 2015.<br>
[DOI](http://dx.doi.org/10.1101/019000) |
Published in Genome Biology 2015.

<b>JP Fortin</b>, A Labbe, M Lemire, BW Zanke, TJ Hudson, EJ Fertig, CMT Greenwood and <b>KD Hansen</b><sup>&dagger;</sup>.<br>
<i>Functional normalization of 450k methylation array data improves replication in large cancer studies.</i><br>
bioRxiv, 2014.<br>
[DOI](http://dx.doi.org/10.1101/002956) |
Published in Genome Biology 2014.

<b>KD Hansen</b>, RA Irizarry, and Z WU<sup>&dagger;</sup>.<br>
<i>Removing technical variability in RNA-Seq data using conditional
quantile normalization.</i><br>
Johns Hopkins University, Dept. of Biostatistics Working
Papers. Working Paper 227.<br>
[URL](http://www.bepress.com/jhubiostat/paper227) |
Published in Biostatistics 2012.

JH Bullard<sup>&lowast;</sup>, EA Purdom<sup>&lowast;</sup>, <b>KD Hansen</b>, S Durinck, and
S Dudoit.<br>
<i>Statistical Inference in mRNA-Seq: Exploratory Data Analysis and
Differential Expression.</i><br>
Working Paper 247, U.C. Berkeley, Division of Biostatistics, 2009.<br>
[URL](http://www.bepress.com/ucbbiostat/paper247/) |
Published in BMC Bioinformatics 2010.

H Bengtsson<sup>&dagger;</sup>, K Simpson, J Bullard, and <b>KD Hansen</b>.<br>
<i>aroma.affymetrix: A generic framework in R for analyzing small to
very large Affymetrix data sets in bounded memory</i>.<br>
Technical Report 745, Department of Statistics, University of
California, Berkeley, 2008.<br>
[PDF](http://www.stat.berkeley.edu/tech-reports/745.pdf) |
Abandoned.

Other
-----

JH Andersen<sup>&dagger;</sup>, M Harhoff, S Grimstrup, I Vilstrup, CF Lassen,
LPA Brandt, AI Kryger, E Overgaard, <b>KD Hansen</b>, and S Mikkelsen.<br>
<i>Computer mouse use predicts acute pain but not prolonged or chronic
pain in the neck and shoulder</i>. <br>
Occup Environ Med 2008, 65(2):126-131.<br>
[DOI](http://dx.doi.org/10.1136/oem.2007.033506) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/17681996)


A-M Nybo Andersen<sup>&dagger;</sup>, <b>KD Hansen</b>, PK Andersen, and
G Davey Smith.<br>
<i>Advanced paternal age and risk of fetal death: a cohort study</i>.<br>
American Journal of Epidemiology 2004, 160(12):1214-1222.<br>
[DOI](http://dx.doi.org/10.1093/aje/kwh332) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/15583374)

H Dan&oslash;<sup>&dagger;</sup>, <b>KD Hansen</b>, P Jensen, JH Petersen, R Jacobsen,
M Ewertz, and E Lynge.<br>
<i>Fertility pattern does not explain social gradient in breast cancer
in Denmark.</i><br>
International Journal of Cancer 2004, 111(3):451-456.<br>
[DOI](http://dx.doi.org/10.1002/ijc.20203) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/15221976)

H Dan&oslash;<sup>&dagger;</sup>, R Jacobsen, <b>KD Hansen</b>, JK Petersen, and
E Lynge.<br>
<i>Use of census data for construction of fertility history for danish
women</i>.<br>
Scandinavian Journal of Public Health 2004, 32:435-441.<br>
[DOI](http://dx.doi.org/10.1080/14034940410028163) |
[PMID](http://www.ncbi.nlm.nih.gov/pubmed/15762028)

K Schultz-Larsen, S Kreiner, S Hanning, N St&oslash;vring,
<b>KD Hansen</b>, and S Lendal.<br>
<i>Den danske &aelig;ldrepleje under forandring ("An Evaluation of the
quality of the danish elder care")</i>. <br>
Govermental report, 2004.

<b>KD Hansen</b> and E Hansen.<br>
<i>Opgaver i videreg&aring;ende sandsynlighedsregning ("Exercises in
Advanced Probability")</i>.<br>
University of Copenhagen (HC&Oslash; Tryk), 2000.

<sup>&lowast;</sup> indicates equal contribution <br>
<sup>&dagger;</sup> indicates corresponding author
